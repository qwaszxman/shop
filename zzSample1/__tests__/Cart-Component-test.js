/*
 * Cart tests for JEST
 *
 * Copyright © Oleg Nosov 2016-Present
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE.txt file in the root directory of this source tree.
 */

import React from 'react';
import renderer from 'react-test-renderer';
import IntlMessageFormat from 'intl-messageformat';
import Enzyme, { mount } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import Cart from '../src/components/Cart/Cart';

Enzyme.configure({ adapter: new Adapter(), });

const testCartLocalization = {
  color: 'Color',
  iPadCase: 'iPad case',
  red: 'Red',
  shoppingCartTitle: 'My lovely shopping cart',
  productName: '{localizedName}',
  quantityLabel: 'Quantity:',
  priceLabel: 'Price:',
  priceValue: '{localizedCurrency}{price}',
  totalLabel: 'Total:',
  totalValue: '{localizedCurrency}{total, plural, ' +
              '=0 {0}' +
              'other {#}}',
  remove: 'Remove',
  productPropertyLabel: '{localizedName}:',
  productPropertyValue: '{localizedValue}',
  GBP: '£',
};

const getLocalization = (id, params = {}) =>
  new IntlMessageFormat(testCartLocalization[id], 'en').format(params);

const createCart = ({ products, }, renderFunc = mount) =>
  renderFunc(<Cart
    products={{ ...products, }}
    onUpdateProduct={
      (key, updatedProduct) => {
          // That's just a test. Don't do like this,
          // use immutable structures
        products[key] = { ...products[key], ...updatedProduct, };
      }
    }
    onRemoveProduct={
        key => void delete products[key]
        // Don't mutate given objects, create new
    }
    checkoutButton={
      <a />
    }
    isCartEmpty={
        false
      }
    getLocalization={
        getLocalization
      }
    currency="GBP"
  />);


describe('Cart', () => {
  const iPadCaseInCart = {
    id: 'ipad-case',
    properties: {
      color: 'red',
    },
    quantity: 1,
    name: 'iPadCase',
    prices: {
      GBP: 70,
    },
    imageSrc: 'ipad-case1-483x321.jpeg',
    propertiesToShowInCart: [ 'color', ],
    path: '/shop/ipad-case',
  };

  it('changes product quantity value', () => {
    // Create a cart with 1 red iPad case in
    const products = { 'ipad-case_red': iPadCaseInCart, };

    const renderedCart = createCart({ products, });

    const quantityInput = renderedCart.find('input');

    // Change iPad case quantity to 10
    quantityInput.instance().value = 10;

    renderedCart.find('input').simulate('change');

    // Now must have 10 cases in cart
    expect(products['ipad-case_red'].quantity).toBe(10);

    quantityInput.instance().value = -1;

    renderedCart.find('input').simulate('change');

    // Expect 10 iPad cases in cart
    expect(products['ipad-case_red'].quantity).toBe(10);

    // Remove our cases from cart
    renderedCart.find('button').simulate('click');

    // Now cart must be empty
    expect(Object.keys(products).length).toBe(0);
  });

  it('takes snapshot', () => {
    const products = { 'ipad-case_red': iPadCaseInCart, };
    const renderedCart = createCart({ products, }, renderer.create);
    expect(renderedCart.toJSON()).toMatchSnapshot();
  });
});
