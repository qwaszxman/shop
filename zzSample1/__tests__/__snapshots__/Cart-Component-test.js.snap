// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Cart takes snapshot 1`] = `
<div
  className="my-1 w-100 list-group animated"
>
  My lovely shopping cart
  <div>
    <div
      className="list-group-item list-group-item-action align-items-start animated"
    >
      <div
        className="row align-items-start"
      >
        <div
          className="col-12 col-sm-12 col-md-4 col-lg-4 col-xl-3 my-1"
        >
          <a
            href="/shop/ipad-case"
          >
            <div
              className="list-group-item-heading"
            >
              iPad case
            </div>
            <img
              alt=""
              className="img-fluid"
              src="ipad-case1-483x321.jpeg"
            />
          </a>
        </div>
        <div
          className="col-12 col-sm-12 col-md-5 col-lg-5 col-xl-7"
        >
          <div
            className="form-group row"
          >
            <label
              className="col-6 col-md-5 col-lg-4 col-form-label"
              htmlFor="quantity"
            >
              Quantity:
            </label>
            <div
              className="col-6 col-md-7 col-lg-8"
            >
              <input
                className="form-control"
                onChange={[Function]}
                type="number"
                value={1}
              />
            </div>
          </div>
          <div
            className="form-group row"
          >
            <div
              className="col-xs-6 col-md-5 col-lg-4"
            >
               
              Color:
            </div>
            <div
              className="col-xs-6 col-md-7 col-lg-8"
            >
               
              Red
            </div>
          </div>
          <div
            className="form-group row"
          >
            <div
              className="col-xs-6 col-md-5 col-lg-4"
            >
               
              Price:
            </div>
            <div
              className="col-xs-6 col-md-7 col-lg-8"
            >
               
              £70
            </div>
          </div>
          <div
            className="form-group row"
          >
            <div
              className="col-xs-6 col-md-5 col-lg-4"
            >
               
              Total:
            </div>
            <div
              className="col-xs-6 col-md-7 col-lg-8"
            >
               
              £70
            </div>
          </div>
        </div>
        <div
          className="col-12 col-sm-12 col-md-3 col-lg-3 col-xl-2"
        >
          <div
            className="form-group row"
          >
            <div
              className="col-12 text-center"
            >
              <button
                className="btn btn-danger form-control"
                onClick={[Function]}
                role="button"
                type="button"
              >
                <i
                  className="fa fa-trash mx-1"
                />
                <span>
                  Remove
                </span>
              </button>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    className="row my-1"
  >
    <div
      className="col-xs-12 col-sm-12 col-md-8 col-lg-6 col-xl-6 m-auto"
    >
      <a />
    </div>
  </div>
</div>
`;
